api/docs/release.dox

```diff
@@ -265,6 +265,12 @@ Further non-compatibility-affecting changes include:
  - Added -trace_instr_intervals_file option to the drmemtrace trace analysis tools
    framework. The file must be in CSV format containing a <start,duration> tracing
    interval per line where start and duration are expressed in number of instructions.
```

Comment:
> In the launcher, if all tools prefer core-sharded, and the user did not specify -no_core_sharded, core-sharded (or core-serial) mode is enabled, 

There's potential for confusion here.

Can you provide more motivation why we need such "automatic" enabling of core-sharded? IMHO the responsibility to (remember to) enable -core_sharded lies with the user.

Reply:
Please read the issue #6949.  The cache and TLB simulators by default are being rather stupid: they do a static scheduling which does not work well at all.  Asking new users to somehow know this and never run our key tools with the defaults sounds like bad UI design: the defaults should do what nearly all users would want.  Users should not have to pass an explicit non-default parameter just to get a tool to operate properly.

Reply:
> Please read the issue #6949. The cache and TLB simulators by default are being rather stupid

I understand the issue, but not yet convinced that this is the correct solution.

> they do a static scheduling which does not work well at all.
> the defaults should do what nearly all users would want

Some may perceive a simpler interleaving to be better. Some may actually want to study thread locality and want thread-shards.

For this problem, there are too many things users may want.

For this problem, It may be better if the defaults do  what users would _expect_ if they don't have in-depth understanding of our system.

Adding the scheduler's complex interleaving will not be expected by most users (as most users probably won't be aware of the scheduler's existence, or the non-trivial scheduler options). It should be up to the aware-user to explore and enable these more sophisticated options.

Reply:
An alternative: if we expect that only these simulators will want default core-sharded, we can potentially create new aliases: drcachesim_core_sharded, tlb_core_sharded, that somehow force the framework to do core-sharded.

Reply:
> Some may perceive a simpler interleaving to be better. Some may actually want to study thread locality and want thread-shards.

But there are no thread shards inside the simulators!  The simulators *must* schedule the threads onto virtual CPUs.  There is scheduling no matter what!  It's a question of whether that scheduling is useful or not.  The default scheduling is broken and not useful or even or representative unless the app's threads are all the same size and make no system calls.

Reply:
> An alternative: if we expect that only these simulators will want default core-sharded, we can potentially create new aliases: drcachesim_core_sharded, tlb_core_sharded, that somehow force the framework to do core-sharded.

I don't think anyone should be running these simulators with their current round-robin static schedule: a separate tool type that still keeps plain drcachesim using the bad naive scheduler does not seem like a solution to me.

Reply:
> that somehow force the framework to do core-sharded

Sounds like it would end up being similar to this PR with a tool able to request that it wants core-sharded...

Reply:
> But there are no thread shards inside the simulators! The simulators must schedule the threads onto virtual CPUs. There is scheduling no matter what!

In the "Some may actually want to study thread locality and want thread-shards." case, perhaps the user will set num_cores == thread_count to have separate stats per thread shard.

> Sounds like it would end up being similar to this PR with a tool able to request that it wants core-sharded...

FWIW I was suggesting that we would hardcode some tools to _always_ run as core-sharded (the ones suffixed with _core_sharded e.g.), instead of letting tools communicate a preference via that API (which introduces more confusion IMHO).

If there are strong opinions on going with the new API, we should atleast change the verbosity level of the clients/drcachesim/analyzer_multi.cpp logs to zero so they're always printed, and in the `switch_to_core_sharded` case also suggest the user to run with `-no_core_sharded` if they didn't want it.

Reply:
> FWIW I was suggesting that we would hardcode some tools to always run as core-sharded (the ones suffixed with _core_sharded e.g.), instead of letting tools communicate a preference via that API (which introduces more confusion IMHO).

Not sure I understand: you mean hardcode handling by name for certain tools we provide, which will not generalize to a third-party user-created tool?  An API seems cleaner and more useful for other tools.

> we should atleast change the verbosity level of the clients/drcachesim/analyzer_multi.cpp logs to zero so they're always printed

We have to be careful with "always printed": that can be very annoying to users if there is no way to disable.  If we made them print at 0 for non-fatal errors we would probably have to add yet another option like -quiet to disable them.

Reply:
I tried always printing and it makes all our tests fail.  I think we have to go back to verbose 1.

Per offline discussion we're keeping the preferred API but having a mix fail.